### PR TITLE
fix(ltm): run context-source fold even when the knowledge table is empty (#961)

### PR DESCRIPTION
**Pull request by BYK**
## The bug
`forSession`'s empty-knowledge fast path returned `[]` as soon as **both** knowledge pools were empty — **before** the context-source fold (#1228/#1293) that surfaces passive distillation/temporal facts:

```js
if (!crossEntries.length && !projectEntries.length) {
  timer.emit("forSession", 0);
  return [];   // ← skips the includeContextSources fold below
}
```

So whenever the `knowledge` table was empty — a fresh project, or **any distillation-only session** — passive injection was silently dead. Facts were captured, embedded, and FTS-matchable, yet never reached the model.

## How it was found
The #961 live eval. On the single-long **lore** arm, the turn-0 aside (`channel=WHOLESALE, region=EMEA, warehouse=WH-07, status=SUBMITTED`) was:
- ✅ captured (distillation row)
- ✅ embedded (`distillation_vec` populated)
- ✅ FTS-matchable (best BM25 rank against the session context)

…yet `forSession(...)` returned **0 entries** and `gateway.log` showed **0** fact mentions. The agent produced generic defaults (`web`/`us-east`/`wh1`/`pending`) and failed the probes — **because Lore handed it nothing, not because the model couldn't apply facts.** (This also means prior single-long lore-arm eval numbers understate Lore — passive injection was off whenever knowledge was empty.)

## The fix
Take the fast path only when the caller did **not** ask for a context-source fold (or there's no session/hint context). When `includeContextSources` is set and a context exists, fall through so the fold runs — the scoring path handles empty knowledge pools fine (`allScored` just starts empty). The fast path is preserved for the common no-context-source empty-knowledge case (verified: 0 entries, no extra embed).

## Test
Existing context-source tests always seeded a knowledge entry, so they never hit the fast path — which is why this slipped through. Added a regression test that seeds a distillation + temporal message with **no knowledge** and asserts the fold still surfaces them, plus that the fast path is preserved when the fold isn't requested. **Mutation-verified:** restoring the unconditional early return turns the test red.

Verified against the actual failing eval DB: with the fix, the `WHOLESALE`/`EMEA`/`WH-07` distillation now surfaces into system[2] (`FACT SURFACED = true`), while the no-context-source empty-knowledge path still returns 0.

### CODE REVIEW

File: packages/core/src/ltm.ts [hunk 1]
```diff
@@ -2485,12 +2485,29 @@ export async function forSession(
...
+  // Fall through so the context-source fold below runs; the scoring path handles
+  // empty knowledge pools fine (allScored simply starts empty).
+  const wantsContextSourceFold =
+    !!options?.includeContextSources?.length &&
+    (!!options.contextHint?.trim() || !!sessionID);
+  if (
+    !crossEntries.length &&
+    !projectEntries.length &&
+    !wantsContextSourceFold
+  ) {
```
[the remark this answers is not in the document]
Comment (BYK) [reply packages/core/src/ltm.ts:2510]: Good catch — valid latent issue. No production caller combines `categories:["preference"]` with `includeContextSources` today (preference reads use `categories:["preference"]` without context sources; context reads use `excludeCategories:["preference"]`), so it isn't reachable now, but the API should stay honest. Fixed in d8253a2: `isPreferenceOnly` now also requires `!wantsContextSourceFold`, so a caller combining the two falls through to the full path and the fold runs. Added a mutation-verified regression test (`preference-category filter + context sources still folds`).
